### DISCUSSION TITLE
Download instructions for dummies?

### DISCUSSION THREAD

**Discussion by selsin** [OvertureMaps/data#62] [Discussion]
Could someone please write dummy-proof instructions on how to download the data, for us who aren't familiar with coding or the coding language?
I have created an AWS account but I have no idea where to paste the link provided in the description s3://overturemaps-us-west-2/release/2023-07-26-alpha.0/theme=places.

I would like to download the data and look at it in QGIS.

Thank you in advance.

**Comment by mtravis** [comment]
@selsin I'm happy try and help you out. 

Firstly you don't need an AWS account to actually get the data. You will need the AWS CLI install on your machine. 

Once you have that you can then download the places data using  this command.

`aws s3 cp --region us-west-2 --no-sign-request --recursive s3://overturemaps-us-west-2/release/2023-07-26-alpha.0/ <DESTINATION>`

Once you have all the data in the destination folder you can convert it to spatial formats using GDAL or DuckDB. 

Can I check to see which type of OS you're using? What you do next kind of determines what advice I can give here.

**Reply by selsin** [reply]
Thanks so much!
Now i installed AWS CLI and I managed to get the themes to start downloading accoridng to your link above. But if I want the data for whole of Sweden what would the link be then?

**Reply by mtravis** [reply]
You can download the geoparquet file for Sweden here: https://beta.source.coop/cholmes/overture/geoparquet-country-quad/country_iso=SE - this is only for the buildings data though. 

In terms of operating system I meant are you using Linux, Windows or OSX?

**Reply by selsin** [reply]
Im using Windows.
If I want to download the other themes, do I just exchange "us-west-2" with "sweden" ?

**Reply by mtravis** [reply]
@selsin `us-west-2` is the AWS region where the data is hosted.   

You get the different themes from here https://github.com/OvertureMaps/data 
Each theme contains all the global data in it's entirety. 

The [source.coop ](https://beta.source.coop/) is an open repo run by Chris Holmes and he has kindly taken the Overture Buildings data and partitioned by country (essentially that means it's been split into countries) That data is stored as geoparquet and QGIS won't open that natively yet (though I'm hoping we can get that to happen soon)

All this is very new and the providers of the open source software need a bit of time to get up to speed 😃

**Reply by selsin** [reply]
Alright, I think I'm following so far.
Could you kindly walk me through how to convert the data using GDAL or DuckDB?

**Reply by mtravis** [reply]
If you have version 3.5+ of GDAL we can try and use that. If not then we can try DuckDB as means you don't have to install conda to get the correct version of GDAL.

**Reply by selsin** [reply]
In my QGIS it says:  GDAL/OGR version 3.6.2

**Reply by mtravis** [reply]
Yes but we aren't using QGIS for this part.

If you run `ogrinfo --formats` do you see Parquet in that list? 

Also try running `ogrinfo --version` to see which version of GDAL you have on the command line.

**Reply by selsin** [reply]
![image](https://github.com/OvertureMaps/data/assets/71225928/81a59a41-d1db-468e-a25a-0b478e478022)

Im not sure if I'm checking in the right place, but if it is it cant find any of them.

**Reply by mtravis** [reply]
I'm pretty sure that when you install QGIS on a Windows machine it comes with a osgeo4W shell? 

Trying looking for that in your menu but if you can't find it then install DuckDB and I'll pass you a script to run.

**Reply by mtravis** [reply]
@selsin have you tried downloading the SE.parquet file here: https://beta.source.coop/cholmes/overture/geoparquet-country-quad/country_iso=SE and seeing if you can open that in QGIS?

**Reply by selsin** [reply]
I have downloaded the parquet file but haven't tried running it in qgis yet. 

I will get back to you when I have done both :)

**Reply by cholmes** [reply]
@selsin - I'm also curious to know if that works on Windows QGIS. I believe it should, if it doesn't try to upload to the latest QGIS. Windows and Linux QGIS should work with GeoParquet. Mac needs to be installed with conda.

And unfortunately I've only converted the buildings layer. I think the others might be easier, so if you want me to try another layer let me know and I can have a go at it. I'm hoping that the core Overture data in the future will be distributed as actual GeoParquet, partitioned geospatially, as it'll make it a lot easier to work with.

**Reply by mtravis** [reply]
@cholmes I've just tested the file on my Linux machine (Pop!_OS 22.04 LTS x86_64) and it won't work with the latest version of QGIS which is 3.32 

![image](https://github.com/OvertureMaps/data/assets/1799238/bbc65dfe-5f7b-4222-8886-7ee1928edc4b)

When I installed and run QGIS in a Conda environment with `conda install qgis libgdal-arrow-parquet` I can open this file (and other gpq files) just fine. I think the standard install must be missing libgdal-arrow-parquet?

**Reply by selsin** [reply]
@cholmes it actually worked on Windows. Im using 3.28.4-Firenze version of Qgis and I just dragged and dropped it :)
If you would like to do it, having the places of interest for Sweden would be great.

**Reply by selsin** [reply]
@mtravis I have now installed DuckDB.

**Reply by cholmes** [reply]
Ah, great to know it worked on Windows - makes me feel more confident recommending that. 

@mtravis I'm curious on Linux if you might be able to install libgdal-arrow-parquet on the normal version? Or does it have to be conda? I can switch the geoparquet.org recommendation to say do conda on linux too. But it'd be nice if it was even easier.

@selsin - I took a go at converting the places - way easier than the buildings stuff (plus I wrote all the scripts I needed already, but still it was much faster to convert 3gb vs like 80gb). Country files are up at https://beta.source.coop/cholmes/overture/places-geoparquet-country - Sweden is at https://data.source.coop/cholmes/overture/places-geoparquet-country/SE.parquet But, uh, opening that causes my qgis to hard crash? Curious what you guys experience on windows and linux. I was able to convert it to shp and gpkg, but not fgb. I put up the gpkg at  https://data.source.coop/cholmes/overture/places-gpkg-country/SE.gpkg

I may switch around these folder names, but anyone in the future should be able to browse to https://beta.source.coop/cholmes/overture and find country partitioned parquet data.

**Reply by selsin** [reply]
I could open both parquet and gpkg files without qgis crashing. 
Thank you so much for the conversion! :)

**Reply by selsin** [reply]
@mtravis I would still like to learn to transform the data using DuckDB, so if you have a chance to pass me the script for that I would be very grateful. I would like to try to download and transform the transportation theme :)

**Reply by mtravis** [reply]
@cholmes It's a know issue with the Flatpak installation of QGIS: https://github.com/flathub/org.qgis.qgis/issues/272 
You can only get recenct version of QGIS on ubuntu by using Conda or Flatpak AFAIK

@selsin sure, no problem. I'll pass something across later.

**Reply by selsin** [reply]
@mtravis hi, sorry to disturb you again. I just wanted to remind you about the script. :)

**Reply by mtravis** [reply]
@selsin No problem. Try the following.

```
INSTALL spatial;
LOAD spatial;

COPY (select * EXCLUDE (geometry), ST_GeomFromWKB(geometry) as geom 
from read_parquet('/data/transportation/type=segment/*') 
where st_intersects(ST_GeomFromWKB(geometry),ST_GeomFromText('POLYGON((10.59 69.06, 24.18 69.06, 24.18 55.14, 10.59 55.14, 10.59 69.06))'))
limit) 
to '/data/transportation/sweden_roads.parquet' (FORMAT PARQUET);
```

Once you have that you'll need to use [gpq](https://github.com/planetlabs/gpq) to convert from parquet to geoparquet

Run this command `gpq convert sweden_roads.parquet sweden_roads_geo.parquet`

Hope that helps

Matt

**Reply by Sok46** [reply]
Can you tell me how to convert parquet to geojson on Windows?

**Reply by selsin** [reply]
Thank you. 
I cant get it to work tho. 
Just to make sure, I have downloaded the transportation theme for the whole world which resulted in two folders: type=connector and type=segment:
Lets say I have these saved on my computer at this location: C:\Users\Selma\Desktop\OvertureMaps\type=segment

Does the code look like this instead then:

INSTALL spatial;
LOAD spatial;

COPY (select * EXCLUDE (geometry), ST_GeomFromWKB(geometry) as geom 
from read_parquet('C:\Users\Selma\Desktop\OvertureMaps\type=segment\*') 
where st_intersects(ST_GeomFromWKB(geometry),ST_GeomFromText('POLYGON((10.59 69.06, 24.18 69.06, 24.18 55.14, 10.59 55.14, 10.59 69.06))'))
limit) 
to 'C:\Users\Selma\Desktop\OvertureMaps\sweden_roads.parquet' (FORMAT PARQUET);

?

**Reply by selsin** [reply]
@Sok46 I will let you know once I have figured it out :)

**Reply by Krizz** [reply]
I've created this easy to use tool to quickly extract GeoJSON/GeoParquet data for different areas from Overture Maps. You might find this useful - or not :)

https://github.com/Krizz/fetch_overture

**Comment by snuup** [comment]
I tried for an hour to download the GeoParquet file holding the Places. I wonder why there is no download link . Would appreciate if you can make it imuch simpler to download the data. For instance OSM data is trivial to download, the planet files are plain http links, please make it as easy as the OSM community does it.
